api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationRequestDaoImpl.java

```diff
@@ -47,13 +53,33 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 					entry.getValue().forEach(d -> handleMedicationReference("d", (ReferenceAndListParam) d.getParam())
 					        .ifPresent(c -> criteria.createAlias("drug", "d").add(c)));
 					break;
+				case FhirConstants.STATUS_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleMedicationRequestStatus((TokenAndListParam) param.getParam())
+					        .ifPresent(criteria::add));
+					break;
 				case FhirConstants.COMMON_SEARCH_HANDLER:
 					handleCommonSearchParameters(entry.getValue()).ifPresent(criteria::add);
 					break;
 			}
 		});
 	}
 	
+	private Optional<Criterion> handleMedicationRequestStatus(TokenAndListParam tokenAndListParam) {
+		return handleAndListParam(tokenAndListParam, token -> {
+			if (token.getValue() != null) {
+				try {
+					return Optional.of(
+					    eq("status", MedicationRequest.MedicationRequestStatus.valueOf(token.getValue().toUpperCase())));
```

Comment:
Is there a `status` field on the OpenMRS model for `DrugOrder`?

Reply:
> Is there a `status` field on the OpenMRS model for `DrugOrder`?

not really , drug order doesn't seem to have a status anywhere on the [data model](https://docs.openmrs.org/datamodel/)